tool/tsh/kube.go

```diff
@@ -662,6 +714,9 @@ func (c *kubeCredentialsCommand) run(cf *CLIConf) error {
 		return trace.Wrap(err)
 	}
 
+	// Unlock and remove the lockfile so subsequent tsh kube credentials calls don't exit early
+	unlockKubeCred(true)
+
 	return c.writeKeyResponse(cf.Stdout(), k, c.kubeCluster)
 }
```

Comment:
Instead of calling the unlock function multiple times, you can just capture the `deleteKubeCredsLock` var using a closure and once it executes, the var is read from pointer before calling `unlockKubeCred`
```go
       var deleteKubeCredsLock = false
	defer func(){
 		unlockKubeCred(deleteKubeCredsLock) 
	}()

	tc, err := makeClient(cf, true)
	if err != nil {
		return trace.Wrap(err)
Expand All
	@@ -628,6 +677,9 @@ func (c *kubeCredentialsCommand) run(cf *CLIConf) error {
		}
		if crt != nil && time.Until(crt.NotAfter) > time.Minute {
			log.Debugf("Re-using existing TLS cert for Kubernetes cluster %q", c.kubeCluster)
			// Unlock and remove the lockfile so subsequent tsh kube credentials calls don't exit early
			deleteKubeCredsLock=true

			return c.writeKeyResponse(cf.Stdout(), k, c.kubeCluster)
		}
		// Otherwise, cert for this k8s cluster is missing or expired. Request
Expand Down
Expand Up
	@@ -662,6 +714,9 @@ func (c *kubeCredentialsCommand) run(cf *CLIConf) error {
		return trace.Wrap(err)
	}

	// Unlock and remove the lockfile so subsequent tsh kube credentials calls don't exit early
	deleteKubeCredsLock=true